PresentData/PresentMonTraceConsumer.cpp

```diff
@@ -1197,6 +1395,11 @@ void PMTraceConsumer::CompletePresent(std::shared_ptr<PresentEvent> p, uint32_t
     if (*presentIter == p) {
         std::lock_guard<std::mutex> lock(mPresentEventMutex);
         while (presentIter != presentDeque.end() && presentIter->get()->Completed) {
+            // TODO: we have to choose between pushing presentIter vs presentDeque[0] here. presentDeque[0] is likely the more reliable one.

```

Comment:
I'd move the assert to replace the comment, and I'd try to be more consistent. `mPresentEvents` is modified using `*presentIter` while `mAllPresents` is modified using `presentDeque[0]`. Those should be the same, regardless of which one it is. In fact, `presentIter` could entirely be deleted and all accesses could be replaced with `presentDeque[0]`, or my preference, `presentDeque.front()`.

I think that'd increase the readability of the below code where we always `pop_front()` even though we have an iterator.